bin/faucet/src/main.rs

```diff
@@ -72,36 +74,42 @@ async fn main() -> Result<(), FaucetError> {
 
             info!(target: COMPONENT, %config, "Initializing server");
 
-            info!("Server is now running on: {}", config.endpoint_url());
-
-            HttpServer::new(move || {
-                let cors = Cors::default().allow_any_origin().allow_any_method();
-                App::new()
-                    .app_data(web::Data::new(faucet_state.clone()))
-                    .wrap(cors)
-                    .wrap(Logger::default())
-                    .wrap(DefaultHeaders::new().add(("Cache-Control", "no-cache")))
-                    .service(get_metadata)
-                    .service(get_tokens)
-                    .service(actix_web_static_files::ResourceFiles::new(
-                        "/",
-                        static_resources::generate(),
-                    ))
-            })
-            .bind((config.endpoint.host, config.endpoint.port))
-            .map_err(|err| FaucetError::StartError(err.to_string()))?
-            .run()
-            .await
-            .map_err(|err| FaucetError::StartError(err.to_string()))?;
+            let app = Router::new()
+                .route("/", get(get_index))
+                .route("/get_metadata", get(get_metadata))
+                .route("/get_tokens", post(get_tokens))
+                .route("/*path", get(get_static_file))
+                .layer(
+                    ServiceBuilder::new()
+                        .layer(TraceLayer::new_for_http())
+                        .layer(SetResponseHeaderLayer::if_not_present(
+                            http::header::CACHE_CONTROL,
+                            HeaderValue::from_static("no-cache"),
+                        ))
+                        .layer(
+                            CorsLayer::new()
+                                .allow_origin(tower_http::cors::Any)
+                                .allow_methods(tower_http::cors::Any),
+                        ),
+                )
+                .with_state(faucet_state);
+
+            let endpoint_url = config.endpoint_url();
+
+            let listener = TcpListener::bind((config.endpoint.host, config.endpoint.port))
+                .await
+                .map_err(|err| FaucetError::StartError(err.to_string()))?;
+
+            info!("Server is now running on: {}", endpoint_url);
```

Comment:
We should prefer structured logs where possible. There are quite a few interpolated logs in the faucet which we should redo.

Might need to add a display implementation to endpoint so that it actually prints something nice like `http://localhost:8080`.

```suggestion
            info!(endpoint=%endpoint, "Server started");
```